src/integer_mod_q/polynomial_ring_zq/from.rs

```diff
@@ -111,3 +179,71 @@ mod test_from_poly_over_z_modulus_polynomial_ring_zq {
         let _ = PolynomialRingZq::from((poly, modulus));
     }
 }
+
+#[cfg(test)]
+mod test_from_str {
+    use super::PolynomialRingZq;
+    use std::str::FromStr;
```

Comment:
you can also call `to_string` on PolynomialRingZq options. Please add a test where you instantiate a PolynomialRingZq, then call `to_string` and then use `from_str` to reinitiate